plugins/net_plugin/net_plugin.cpp

```diff
@@ -2830,9 +2846,11 @@ namespace eosio {
 
    size_t calc_trx_size( const packed_transaction_ptr& trx ) {
       // transaction is stored packed and unpacked, double packed_size and size of signed as an approximation of use
+      const auto cfd_size = trx->get_context_free_data() != nullptr ? trx->get_context_free_data()->size() : 0;
```

Comment:
This is the number of context free data segments, which is probably not what you're looking for.

Reply:
Added an `estimated_size` to `packed_transaction`, used that instead.